sandbox_dns_unix.go

```diff
@@ -81,7 +81,7 @@ func (sb *sandbox) buildHostsFile() error {
 	}
 
 	// This is for the host mode networking
-	if sb.config.originHostsPath != "" {
+	if sb.config.useDefaultSandBox && len(sb.config.extraHosts) == 0 {
 		if err := copyFile(sb.config.originHostsPath, sb.config.hostsPath); err != nil && !os.IsNotExist(err) {
```

Comment:
nit -- assumes that if the condition holds true that originHostsPath has also been set.  This is true in the moby patch of course, but is unclear from just reading the libnetwork code.  Suggest at least putting a comment on OptionUseDefaultSandbox() to say that one must also add OptionOriginHostsPath() or one or more OptionExtraHost() options or expect an error.  Same for OptionOriginResolvConfPath() and OptionDNS()/OptionDNSSearch()/OptionDNSOptions()...